zipline/data/data_portal_live.py

```diff
@@ -49,20 +49,28 @@ def get_history_window(self,
         # get_spot_value() will be used to fill the missing data - which is
         # always representing the current spot price presented by Broker.
 
-        historical_bars = super(DataPortalLive, self).get_history_window(
-            assets, end_dt, bar_count, frequency, field, data_frequency,
-            ffill=False)
-
         realtime_bars = self.broker.get_realtime_bars(
             assets, frequency)
 
+        try:
+            historical_bars = super(DataPortalLive, self).get_history_window(
+                assets, end_dt, bar_count, frequency, field, data_frequency,
+                ffill=False)
+        except:
+            historical_bars = None
```

Comment:
I'm not exactly sure what is the motivation behind the logic here.
Are we indirectly leveraging Alpaca's data api to fetch historical bars?

If so, it might negatively affect the behavior when one is using IB as we don't fetch historical bars using that API.

We need at least some documentation why this is needed (use-case) and make sure that IB API does not get affected.

Additionally: the exception is too broad here, it would help the understanding if we'd match for an exact exception.

Reply:
This change is in fact not much related to Alpaca broker. Here is what I went through: I had quantopian-quandl bundle (daily bar) and tested intraday behavior using Alpaca. Of course, the bundle doesn't have any minute bars data so it raised an exception here (I think it was "no such file found"). I am pulling only a few hundreds of bars in minute level and totally find not having longer history, so I patched this way to fall back to rely on only Alpaca's data API.

This change is really only to fall back to use only broker API data *in case* bundle doesn't have it.  It would run the same way if you have enough data in bundle.  It doesn't really matter which broker we are talking about.

I am totally fine reverting this change if you want and discuss further separately, or just add comment here to explain this scenario.

Reply:
I understand why you wanted this and there are thoughts about how to continuously ingest but the current architecture is not really built for it as the asset_finder object is used everywhere based o the ingested bundle and the main rule is that it needs to be ingested previously. This change would be great if we make sure we actually insert the asset in the bundle but this is a far bigger change. Soon we will have a standard IEX bundle with at least some minute data in it so that will mitigate this except for the most exotic tickers.

Reply:
Sure, I'm glad to hear we are on the same page to fix the fundamental issues here.

As a user (I am writing some algo using zipline-alpaca), I am now facing how to resolve this issue.  The only solution is to build my own minute level bundle, but I guess that should be commonly shared, otherwise not many people can trade with zipline?